polkadot/runtime/parachains/src/assigner_on_demand/migration.rs

```diff
@@ -67,7 +67,7 @@ mod v1 {
 					Pallet::<T>::add_on_demand_order(
 						queue_status,
 						enqueued_order.para_id,
-						QueuePushDirection::Front,
+						QueuePushDirection::Back,
```

Comment:
Maybe it would be a good idea to add a [migration test](https://github.com/paritytech/polkadot-sdk/pull/3479/files#diff-8247e57109bbaa6dd0f7271784ad785d3240a2e72c081ddf7c658c08deedde85R143), to make sure we are preserving the order correctly.